packages/react/src/Banner/Banner.tsx

```diff
@@ -0,0 +1,323 @@
+    }
+  }, [titleId])
+  const dismissible = variant !== 'critical' && onDismiss
+  const hasActions = primaryAction || secondaryAction
+
+  if (__DEV__) {
+    // Note: __DEV__ will make it so that this hook is consistently called, or
+    // not called, depending on environment
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+      const title = document.getElementById(titleId)
+      if (!title) {
+        throw new Error(
+          'The Banner component requires a title to be provided as the `title` prop or through `Banner.Title`',
+        )
+      }
+    }, [titleId])
+  }
+
+  return (
+    <BannerContext.Provider value={value}>
+      <StyledBanner {...rest} aria-labelledby={titleId} as="section" data-variant={variant} tabIndex={-1} ref={ref}>
+        <div className="BannerIcon">{icon && variant === 'info' ? icon : iconForVariant[variant]}</div>
+        <div className="BannerContainer">
+          <div className="BannerContent">
+            {title ? <BannerTitle>{title}</BannerTitle> : null}
+            {description ? <BannerDescription>{description}</BannerDescription> : null}
+            {children}
+          </div>
+          {hasActions ? <BannerActions primaryAction={primaryAction} secondaryAction={secondaryAction} /> : null}
+        </div>
+        {dismissible ? (
+          <IconButton
+            aria-label="Dismiss banner"
+            onClick={onDismiss}
+            className="BannerDismiss"
+            icon={XIcon}
+            variant="invisible"
+          />
+        ) : null}
+      </StyledBanner>
+    </BannerContext.Provider>
+  )
+})
+
+/**
+ * For styling, it's important that the icons and the text have the same height
+ * for alignment to occur in multi-line scenarios. Currently, we use a
+ * line-height of `20px` so that means that the height of icons should match
+ * that value.
+ */
+const StyledBanner = styled.div`
+  display: grid;
+  grid-template-columns: auto minmax(0, 1fr) auto;
+  align-items: start;
+  background-color: var(--banner-bgColor);
+  border: 1px solid var(--banner-borderColor);
+  padding: 0.5rem;
+  border-radius: ${get('radii.2')};
+
+  &[data-variant='critical'] {
+    --banner-bgColor: ${get('colors.danger.subtle')};
+    --banner-borderColor: ${get('colors.danger.muted')};
+    --banner-icon-bgColor: ${get('colors.danger.fg')};
+  }
+
+  &[data-variant='info'] {
+    --banner-bgColor: ${get('colors.accent.subtle')};
+    --banner-borderColor: ${get('colors.accent.muted')};
+    --banner-icon-bgColor: ${get('colors.accent.fg')};
+  }
+
+  &[data-variant='success'] {
+    --banner-bgColor: ${get('colors.success.subtle')};
+    --banner-borderColor: ${get('colors.success.muted')};
+    --banner-icon-bgColor: ${get('colors.success.fg')};
+  }
+
+  &[data-variant='upsell'] {
+    --banner-bgColor: ${get('colors.done.subtle')};
+    --banner-borderColor: ${get('colors.done.muted')};
+    --banner-icon-bgColor: ${get('colors.done.fg')};
+  }
```

Comment:
Once https://github.com/primer/react/pull/4491 merges, you can access `upsell` colors. However, these are just CSS variables and won't really have a fallback. If you want to keep using the `get` configuration, you can just use the `done` colors as the fallback. Otherwise maybe we just move to pure CSS variables? 

```
--fgColor-upsell
--bgColor-upsell-muted
--borderColor-upsell-muted
```